src/Block/Js.php

```diff
@@ -45,32 +63,40 @@ public function getOfflinePageUrl()
     }
 
     /**
-     * Get the list of URLs unavailable for caching or viewing offline.
+     * Get the list of URLs for external scripts to import into the service worker.
      *
-     * @return array
+     * @return string[]
      */
-    public function getUrlBlacklist()
+    public function getExternalScriptUrls()
     {
-        return $this->config->getUrlBlacklist();
+        $scripts = [
+            $this->getViewFileUrl("Meanbee_ServiceWorker::js/lib/workbox-sw.prod.v1.0.1.js"),
+        ];
+
+        if ($this->isGaOfflineEnabled()) {
+            $scripts[] = $this->getViewFileUrl("Meanbee_ServiceWorker::js/lib/workbox-google-analytics.prod.v1.0.0.js");
```

Comment:
If this method is made more generic in the future (perhaps to accommodate #4) then some validation should be done on the contents of `$scripts` to make sure a URL was successfully returned from `$this->getViewFileUrl`.